subsys/bluetooth/controller/ll_sw/ull_conn_types.h

```diff
@@ -31,7 +31,6 @@ enum llcp {
 };
 
 struct ll_conn {
-	struct evt_hdr  evt;
```

Comment:
Removal of this struct makes the `ull_hdr` the first item. Because elements of type `struct ll_conn` reside in a memory pool, the `mem_init` initially uses the first 32-bit as an internal pointer. Previously this was the memory occupied by `evt_hdr->ticks_xtal_to_start`, which is always assigned before use. Now the memory shared with the internal pointer is  `ull_hdr->ref` (and unused padding bytes). This is fatal because the member is never initialized.
The problem can be fixed by setting `hdr->ref = 0` in `ull_hdr_init`.

Reply:
Awesome catch!

Reply:
Good catch. The changes in this PR has disclosed many out of bound and uninitialized memory accesses that existed. Will add the initialization and update this PR.